be/src/exprs/quantile_function.cpp

```diff
@@ -0,0 +1,127 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exprs/quantile_function.h"
+#include "exprs/anyval_util.h"
+#include "gutil/strings/numbers.h"
+#include "gutil/strings/split.h"
+#include "util/quantile_state.h"
+#include "util/string_parser.hpp"
+#include "util/slice.h"
+
+namespace doris {
+
+using doris_udf::DoubleVal;
+using doris_udf::StringVal;
+using doris_udf::FloatVal;
+
+void QuantileStateFunctions::init(){}
+
+void QuantileStateFunctions::quantile_state_init(FunctionContext* ctx, StringVal* dst) {
+    dst->is_null = false;
+    dst->len = sizeof(QuantileState<double>);
+    dst->ptr = (uint8_t*) new QuantileState<double>();
+}
+
+static StringVal serialize(FunctionContext* ctx, QuantileState<double>* value) {
+    StringVal result(ctx, value->get_serialized_size());
+    value->serialize(result.ptr);
+    return result;
+}
+
+StringVal QuantileStateFunctions::to_quantile_state(FunctionContext* ctx, const StringVal& src) {
+    QuantileState<double> quantile_state;
+    quantile_state.set_compression(2048);
```

Comment:
It's better to use a named constant to represent `2048` and `10000`, etc, rather than these magic numbers.